### DISCUSSION TITLE
Came up with a catchy title

### DISCUSSION THREAD

**Discussion by Sebosek** [dotnet/maui#11475] [General]
I've tried to develop a very small **mobile** app in the .NET MAUI to get the vibe and understand the hype. But, I have to admit, that I always thought that MAUI should be the successor of Xamarin, primarily focused on mobile development. I was wrong.

Let me tell a story. I am originally a full-stack developer from the times when we weren't called full-stack developers, we were just software developers or software engineers. I develop single-page apps in TypeScript, which is really really great, all the credit to Microsoft for it. At the backend, I develop services running in .NET written in C#. I have 11 years of experience, but still consider myself young. In the last year and a half, I've started with mobile development, the first app is in Xamarin, a personal time-tracker. Nothing big, nothing small (Xamarin.Essentials, Microsoft Toolkit, .NET Core dependency container, Entity Framework, Polly, Refit, SkiaSharp, QR Code scanner, basically I took cloud-based N-layer architecture and put it inside a mobile app), and second is in React Native - a month ago we had our first release, it's an app for a non-profit organization.

The third app, I've created because I like board games, Everdell is great. I had some time to spare, so I decided to develop an app for counting scores. Small app™ using the Reactive Extensions, another great product, developed inside Microsoft, but unfortunately doesn't get the credit it deserves. In the MAUI.

I don't know why, but I expected that all the pain points that I have with Xamarin would be fixed there. To be transparent, here is a list:

- Disposing of resources of abandoned pages (The dispose pattern for pages)
- Improve/fix VisualStateManager (Have you ever tried to style (un)checked disabled radio box?)
- Simplify the creation of custom controls and their binding (source generators, maybe?)

But, MAUI has not focused on them, instead, they decided to focus on something else. Platform renderers, a single project, improving working with images, and many more. Simply put, I don't really care about them, yet.

I've never used platform renderers because simply I didn't need to. And when I needed it, I used SkiaSharp and custom controls. And the single project is nice, right, but I would be equally happy with better MSBuild docs. But, not saying that both are bad, just saying I don't really care.

But, even though the app was really small, I wasn't able to make it work in MAUI, or it would require a lot of investments to find workarounds. I hit:

- Unable to render custom content in radio button,
- Unable to change the color of the back button,
- Plus the same problems that I already know from Xamarin.

On the other hand, web component binding may be tempting. I can imagine that there is a team, out there, within a company that starts a green field project, with big plans of having a web app, mobile app, and maybe even internal tools for its own employees as a desktop app with a progressive PM who see the benefits of using MAUI. I guess that's state-of-the-art of MAUI.

Sure, it may work. But being in the shoes of already spoken PM I would probably hire 2 backend developers and 2 frontend developers instead of 4 csharpists. Simply because from my experience, I don't know many csharpists who know how to scale CSS frameworks or components with good design sense, someone who knows UI patterns and can productively cooperate with UX. Somehow, I more trust that I would be more lucky with React/Angular/Vue developer. Just a personal opinion; however, I would be interested to hear why Microsoft is going in this direction, even though everyone knows the story of how Cortana ended up. Maybe dogfooding, the Office, and Teams are apps based on web technologies, may it be a reason?

I am not complaining, I'm sure there are plenty of happy developers, who are excited about what we get and what's coming.

What is your feeling?
What were your expectations?
And most importantly, are you happy with MAUI overall?

**Comment by akhanalcs** [comment]
That was an interesting observation. 
If Microsoft makes developing beautiful UIs in MAUI easier than the competition, dogfood MAUI in their own apps, squash the bugs faster, I think MAUI has a good future. 

I believe Microsoft dogfooding MAUI will determine the future of MAUI. That part is incredibly important for the success of MAUI.
